dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ConnectionPoolHealthCheck.java

```diff
@@ -2,31 +2,93 @@
 
 import com.codahale.metrics.Gauge;
 import com.codahale.metrics.health.HealthCheck;
+import io.dockstore.webservice.DockstoreWebserviceConfiguration.ExternalConfig;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
+import software.amazon.awssdk.services.cloudwatch.model.MetricDatum;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataRequest;
+import software.amazon.awssdk.services.cloudwatch.model.StandardUnit;
 
 public class ConnectionPoolHealthCheck extends HealthCheck  {
+
     private static final Logger LOG = LoggerFactory.getLogger(ConnectionPoolHealthCheck.class);
 
     private final int maxConnections;
     private final Map<String, Gauge> metricGauges;
+    private CloudWatchClient cw = null;
+    private String namespace = null;
 
-    public ConnectionPoolHealthCheck(int maxConnections, Map<String, Gauge> metricGauges) {
+    public ConnectionPoolHealthCheck(ExternalConfig configuration, int maxConnections, Map<String, Gauge> metricGauges) {
         this.maxConnections = maxConnections;
         this.metricGauges = metricGauges;
+
+        this.cw = CloudWatchClient.builder()
+            .credentialsProvider(DefaultCredentialsProvider.create())
+            .build();
```

Comment:
FWIW, last sprint, I researched the metrics submission mechanisms a little bit, and there at least two others:

* Write metric JSON to logs and have cloudwatch agent assimilate it: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format.html
* Bind cloudwatch agent to a port, and send metric JSON to it as messages.

Pros to these approaches would be that they are more loosely bound to AWS, cons are that they require more code.  I personally like the log-based approach because the metrics appear in the logs and we can see them during development, even if we're not linked to aws.

Reply:
> Pros to these approaches would be that they are more loosely bound to AWS, cons are that they require more code. 

FWIW, as discussed earlier, I'd rather avoid the logs for things that are purely numbers. 
That said, I think if you really needed the numbers I think I can add this in the follow-up more kosher dropwizard metrics follow-up ticket (i.e. add a periodic ConsoleReporter) rather than go through AWS stuff like the embedded metric format if this is for when AWS is not in the picture anyway

Reply:
> > Pros to these approaches would be that they are more loosely bound to AWS, cons are that they require more code.
> 
> FWIW, as discussed earlier, I'd rather avoid the logs for things that are purely numbers. That said, I think if you really needed the numbers I think I can add this in the follow-up more kosher dropwizard metrics follow-up ticket (i.e. add a periodic ConsoleReporter) rather than go through AWS stuff like the embedded metric format if this is for when AWS is not in the picture anyway

Not sure how it actually works, but it's possible that the cloudwatch agent doesn't put the embedded metrics into the log stream, but instead parses them and submits them to Cloudwatch as Metrics.